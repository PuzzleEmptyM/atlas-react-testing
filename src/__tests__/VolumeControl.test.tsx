import { render, fireEvent } from '@testing-library/react';
import VolumeControl from '../components/VolumeControl';

describe('VolumeControl Component', () => {
  it('renders with default volume', () => {
    const { asFragment } = render(<VolumeControl />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders when muted', () => {
    const { getByRole, asFragment } = render(<VolumeControl />);
    const slider = getByRole('slider') as HTMLInputElement;
    fireEvent.change(slider, { target: { value: '0' } });
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders at maximum volume', () => {
    const { getByRole, asFragment } = render(<VolumeControl />);
    const slider = getByRole('slider') as HTMLInputElement;
    fireEvent.change(slider, { target: { value: '100' } });
    expect(asFragment()).toMatchSnapshot();
  });
});
