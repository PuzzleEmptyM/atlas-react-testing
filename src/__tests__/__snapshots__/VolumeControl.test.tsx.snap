// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControl Component > renders at maximum volume 1`] = `
<DocumentFragment>
  <div
    class="flex items-center space-x-2 mt-6"
  >
    <img
      alt="Volume"
      class="w-5 h-5"
      src="/src/assets/medium-volume.png"
    />
    <input
      class="w-48 p-1 h-2 rounded-lg appearance-none cursor-pointer bg-gray-20 range-sm"
      id="small-range"
      max="100"
      min="0"
      style="border: 1px solid black;"
      type="range"
      value="100"
    />
  </div>
</DocumentFragment>
`;

exports[`VolumeControl Component > renders when muted 1`] = `
<DocumentFragment>
  <div
    class="flex items-center space-x-2 mt-6"
  >
    <img
      alt="Mute"
      class="w-5 h-5"
      src="/src/assets/mute.png"
    />
    <input
      class="w-48 p-1 h-2 rounded-lg appearance-none cursor-pointer bg-gray-20 range-sm"
      id="small-range"
      max="100"
      min="0"
      style="border: 1px solid black;"
      type="range"
      value="0"
    />
  </div>
</DocumentFragment>
`;

exports[`VolumeControl Component > renders with default volume 1`] = `
<DocumentFragment>
  <div
    class="flex items-center space-x-2 mt-6"
  >
    <img
      alt="Volume"
      class="w-5 h-5"
      src="/src/assets/medium-volume.png"
    />
    <input
      class="w-48 p-1 h-2 rounded-lg appearance-none cursor-pointer bg-gray-20 range-sm"
      id="small-range"
      max="100"
      min="0"
      style="border: 1px solid black;"
      type="range"
      value="50"
    />
  </div>
</DocumentFragment>
`;
